ngraph/core/shape_inference/include/batch_to_space_shape_inference.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstdint>
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/batch_to_space.hpp>
+#include <openvino/opsets/opset2.hpp>
+
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v1 {
+
+template <class T>
+void shape_infer(const ov::op::v1::BatchToSpace* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 4 && output_shapes.size() == 1);
+    const auto& data_shape = input_shapes[0];
+    const auto& block_shape = input_shapes[1];
+    const auto& crops_begin_shape = input_shapes[2];
+    const auto& crops_end_shape = input_shapes[3];
+
+    auto inputs_same_ps = crops_begin_shape;
+    NODE_VALIDATION_CHECK(op,
+                          T::merge_into(inputs_same_ps, crops_end_shape) && T::merge_into(inputs_same_ps, block_shape),
+                          "block_shape, crops_begin and crops_end inputs must have the same shape. Got: ",
+                          block_shape,
+                          ", ",
+                          crops_begin_shape,
+                          " and ",
+                          crops_end_shape);
+
+    const ov::Rank inputs_rank_one = inputs_same_ps.rank();
+    NODE_VALIDATION_CHECK(op,
+                          inputs_rank_one.compatible(1),
+                          "block_shape and crops inputs must have rank 1. Got: ",
+                          inputs_rank_one);
+
```

Comment:
unused variable. also it could make problems if we would switch to different return value for StaticShape
```suggestion
    NODE_VALIDATION_CHECK(op,
                          inputs_same_ps.rank().compatible(1),
                          "block_shape and crops inputs must have rank 1. Got: ",
                          inputs_rank_one);

```